server/src/main/java/org/elasticsearch/index/codec/PerFieldMapperCodec.java

```diff
@@ -84,6 +89,25 @@ public KnnVectorsFormat getKnnVectorsFormatForField(String field) {
 
     @Override
     public DocValuesFormat getDocValuesFormatForField(String field) {
+        if (useTSDBDocValuesFormat(field)) {
+            return tsdbDocValuesFormat;
+        }
         return docValuesFormat;
     }
+
+    private boolean useTSDBDocValuesFormat(final String field) {
+        return isTimeSeriesModeIndex() && isNotSpecialField(field) && isNumericType(field);
+    }
+
+    private boolean isTimeSeriesModeIndex() {
+        return IndexMode.TIME_SERIES.equals(mapperService.getIndexSettings().getMode());
+    }
+
+    private boolean isNumericType(String field) {
+        return mapperService != null && mapperService.mappingLookup().getMapper(field) instanceof NumberFieldMapper;
```

Comment:
The current logic doesn't include the `@timestamp` field, because it extends from `DateFieldMapper`.
Maybe make an exception for `@timestamp` here? For this field the we know the date is always increasing.